cmd/kubeadm/app/cmd/phases/preflight.go

```diff
@@ -45,21 +65,36 @@ type preflightMasterData interface {
 	IgnorePreflightErrors() sets.String
 }
 
+type preflightJoinData interface {
+	Cfg() *kubeadmapi.JoinConfiguration
+	InitCfg() (*kubeadmapi.InitConfiguration, error)
+	IgnorePreflightErrors() sets.String
+}
+
 // NewPreflightMasterPhase creates a kubeadm workflow phase that implements preflight checks for a new master node.
 func NewPreflightMasterPhase() workflow.Phase {
 	return workflow.Phase{
 		Name:    "preflight",
 		Short:   "Run master pre-flight checks",
 		Long:    "Run master pre-flight checks, functionally equivalent to what implemented by kubeadm init.",
-		Example: masterPreflightExample,
+		Example: initPreflightExample,
 		Run:     runPreflightMaster,
 		InheritFlags: []string{
 			options.CfgPath,
 			options.IgnorePreflightErrors,
+			options.APIServerAdvertiseAddress,
+			options.APIServerBindPort,
+			options.NodeCRISocket,
+			options.NodeName,
+			options.FileDiscovery,
```

Comment:
Fixing args for Init phases is a little bit out of the scope of this PR.
However, if you want to do this, you should consider that you can Inherit only flags that are part of the parent command, and I'm pretty sure init doesn't have all the discovery flags...

Reply:
agreed, i think was mental typo